yammer-extra/src/main/java/com/arpnetworking/metrics/yammer/MetricsRegistry.java

```diff
@@ -0,0 +1,240 @@
+                new Closer(_lock, metricsFactory, _openMetrics, this),
+                CLOSER_PERIOD_MILLIS,
+                CLOSER_PERIOD_MILLIS,
+                TimeUnit.MILLISECONDS);
+    }
+
+    /**
+     * Public constructor.
+     *
+     * @param clock    a {@link Clock} instance
+     */
+    public MetricsRegistry(final Clock clock) {
+        // TODO(barp): Read the settings from a config file [AINT-639]
+        this(
+                new TsdMetricsFactory.Builder()
+                        .setSinks(Collections.singletonList(new TsdQueryLogSink.Builder().build()))
+                        .build(),
+                clock);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Counter newCounter(final MetricName metricName) {
+        final Counter counter = getOrCreate(metricName.getName(), _counterBuilder);
+        getOrAdd(metricName, counter);
+        return counter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Histogram newHistogram(final MetricName metricName, final boolean biased) {
+        final Histogram histogram = getOrCreate(metricName.getName(), (n) -> new Histogram(n, _lock, biased));
+        getOrAdd(metricName, histogram);
+        return histogram;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Timer newTimer(
+            final MetricName metricName,
+            final TimeUnit durationUnit,
+            final TimeUnit rateUnit) {
+        final Timer timer = getOrCreate(metricName.getName(), (n) -> new Timer(n, _lock, tickPool(), durationUnit, rateUnit, _clock));
+        getOrAdd(metricName, timer);
+        return timer;
+
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Meter newMeter(final MetricName metricName, final String eventType, final TimeUnit unit) {
+        final Meter meter = getOrCreate(metricName.getName(), (n) -> new Meter(n, _lock, tickPool(), eventType, unit, _clock));
+        getOrAdd(metricName, meter);
+        return meter;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public <T> Gauge<T> newGauge(final MetricName metricName, final Gauge<T> metric) {
+        _gauges.put(metricName.getName(), metric);
+        getOrAdd(metricName, metric);
+        return metric;
+    }
+
+    @Nonnull
+    /**
+     * Writes the gauges to the metric instance.
+     *
+     * NOTE: This is public to allow use from the Closer when shaded.
+     */
+    public void writeGauges(final Metrics metrics) {
+        try {
+            for (final Map.Entry<String, Gauge<?>> entry : _gauges.entrySet()) {
+                final Object value = entry.getValue().value();
+                if (value instanceof Number) {
+                    metrics.setGauge(entry.getKey(), ((Number) value).doubleValue());
+                }
+            }
+            //CHECKSTYLE.OFF: IllegalCatch - we need to catch everything
+        } catch (final Throwable t) {
```

Comment:
I thought he current wisdom was to catch just Exception. Anything else should be considered disastrous.